java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture3/tasks/_3/Number3.java

```diff
@@ -0,0 +1,199 @@
+package com.zoolatech.lecture3.tasks._3;
+
+/*
+Create an hierarchy of 3 classes:
+Circle, Triangle, Rectangle, and a basic interface
+Shape that provides methods to find a perimeter and an area for each shape.
+In the main method, create a list of three objects and display their perimeter and area.
+Then display the radius of a circle and lengths of a triangle and rectangle sides for objects stored in the list.
+Prevent other classes from extension from the Shape interface.
+It should be also possible to check if two object references of these types
+are the same and to print the inner state of an object (like radius or width)
+to the console output when an object is passed to the System.out.println method.
+*/
+
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.Scanner;
+
+public class Number3 {
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        ArrayList<Shape> shapes = new ArrayList<>();
+
+        System.out.println("Input Circles Radius");
+        Circle circle = new Circle(scanner.nextDouble());
+        shapes.add(0, circle);
+        System.out.println("Circle radius is: " + circle.getRadius() + "\n");
+
+        System.out.println("Input Triangle 3 sides in a row:");
+        Triangle triangle = new Triangle(scanner.nextDouble(), scanner.nextDouble(), scanner.nextDouble());
+        shapes.add(1, triangle);
+        System.out.println("Triangle sides are: " + triangle + "\n");
+
+        System.out.println("Input Width and Length of a Rectangle:");
+        Rectangle rectangle = new Rectangle(scanner.nextDouble(), scanner.nextDouble());
+        shapes.add(2, rectangle);
+        System.out.println("Rectangle: " + rectangle + "\n");
+
+        for (Shape shape : shapes) {
+            String string = shape.toString();
+            System.out.println("List have this Shapes:");
+            System.out.println(string + "\n");
+        }
+        System.out.println();
+
+        for (Shape shape : shapes) {
+            System.out.println("Circle Perimeter: " + circle.getPerimeter() + "\n" + "Circle Area: " + circle.getArea());
```

Comment:
you need to use a variable `shape` to output the perimeter and area, and not specific variables for these shapes